cli-output/src/cli_output.rs

```diff
@@ -378,22 +390,45 @@ impl fmt::Display for CliValidators {
             ))
             .bold()
         )?;
-        for validator in &self.current_validators {
-            write_vote_account(
-                f,
-                validator,
-                self.total_active_stake,
-                self.use_lamports_unit,
-                false,
-            )?;
+
+        let mut sorted_validators = self.validators.clone();
+        match self.validators_sort_order {
+            CliValidatorsSortOrder::Delinquent => {
+                sorted_validators.sort_by_key(|a| a.delinquent);
+            }
+            CliValidatorsSortOrder::Commission => {
+                sorted_validators.sort_by_key(|a| a.commission);
+            }
+            CliValidatorsSortOrder::Credits => {
+                sorted_validators.sort_by_key(|a| a.credits);
+            }
+            CliValidatorsSortOrder::Identity => {
+                sorted_validators.sort_by(|a, b| a.identity_pubkey.cmp(&b.identity_pubkey));
+            }
+            CliValidatorsSortOrder::LastVote => {
+                sorted_validators.sort_by_key(|a| a.last_vote);
+            }
+            CliValidatorsSortOrder::Root => {
+                sorted_validators.sort_by_key(|a| a.root_slot);
+            }
+            CliValidatorsSortOrder::VoteAccount => {
+                sorted_validators.sort_by(|a, b| a.vote_account_pubkey.cmp(&b.vote_account_pubkey));
+            }
+            CliValidatorsSortOrder::Stake => {
+                sorted_validators.sort_by_key(|a| a.activated_stake);
```

Comment:
Could do this to avoid stake order breakage, no?

```suggestion
                sorted_validators.sort_by(|a,b|| a.activated_stake.cmp(&b.activated_stake).reverse());
```

Reply:
Yes, I did that initially actually. But then I discovered that the higher staked nodes at the end by default felt better to me, since I didn't have to scroll allllllll the way up to find the ones that matter. So I'm kinda not sorry for the change here, but wdyt?

Reply:
Oh yeah, that is nice actually.  Maybe just reprint the header as a footer? :wink:

Reply:
ha, yes.  I pushed a commit for this.  > 100 nodes will cause it to print the header as a footer as well.  Here's what the tail of the default MB output now looks like:
```
580  LunaowJnt875WWoqDkhHhE93SNYHa6tfFNVn1rqc57c   LunaFpQkZsZVJL2P2BUqNDJqyVYqrw9buQnjQtMLXdK     8%   74413453    74413421         88645  1.5.18  4698103.30179021 SOL (1.52%)
581  DDnAqxJVFo2GVTujibHt5cjevHMSE9bo8HJaydHoshdp  9GJmEHGom9eWo4np4L5vC6b6ri1Df2xN8KFoWixvD1Bs   10%   74413452    74413420         88730  1.5.18  4761147.227427287 SOL (1.54%)
582  binance (tDw8..MsWp)                          47e2BfrzbWdDieFZA7fc9fX1f73WTtkok2fQxSb8GueY    8%   74413449    74413416         89435  1.5.18  5063308.428957816 SOL (1.63%)
583  melea (6TkK..e1ea)                            GvZEwtCHZ7YtCkQCaLRVEXsyVvQkRDhJhQgB6akPme1e    0%   74413452    74413420         88238  1.5.18  5298336.939869743 SOL (1.71%)
584  Certusm1sa411sMpV9FPqU5dXAYhmmhygvxJ23S6hJ24  CertusDeBmqN8ZawdkxK5kFGMwBXdudvWHYwtNgNhvLu   10%   74413453    74413421         88698  1.5.18  5381906.431079129 SOL (1.74%)
585  8WjzRz9HM2bWBopD6UPbi3swyRE5kk9aJojhn6fBKTJk  9c5bpzVRbfsYY2fannb4hyX5CJUPg3BfH2cL6sR7kJM4    5%   74413452    74413420         88590  1.5.18  5528173.164367185 SOL (1.78%)
586  Fd7btgySsrjuo25CJCj7oE7VPMyezDhnx7pZkj2v69Nk  CcaHc2L43ZWjwCHART3oZoJvHLAe9hzT2DJNUpBzoTN1    7%   74413452    74413420         88136  1.5.18  6050046.482880585 SOL (1.95%)
587  XkCriyrNwS3G4rzAXtG5B1nnvb5Ka1JtCku93VqeKAr   beefKGBWeSpHzYBHZXwp5So7wdQGX6mu4ZHCsH3uTar     8%   74413452    74413420         88244  1.5.18  6842565.525302373 SOL (2.21%)
588  Dokia75SVtetShgapUBoVFfYjL99fQyr1twxKKyTZKa3  H3GhqPMwvGLdxWg3QJGjXDSkFSJCsFk3Wx9XBTdYZykc    5%   74413452    74413420         87860  1.5.18  7261844.753157083 SOL (2.34%)
589  DWvDTSh3qfn88UoQTEKRV2JnLt5jtJAVoiCo3ivtMwXP  FKsC411dik9ktS6xPADxs4Fk2SCENvAiuccQHLAPndvk    7%   74413452    74413420         87848  1.5.18  11042403.16188154 SOL (3.56%)
590  EvnRmnMrd69kFdbLMxWkTn1icZ7DCceRhvmb2SJXqDo4  9QU2QSxhb24FUX3Tu2FpczXjpK3VYrvRudywSZaM29mF    7%   74413453    74413421         86616  1.5.18  11915401.61893568 SOL (3.84%)
591  Awes4Tr6TX8JDzEhCZY2QVNimT6iD1zWHzf1vNyGvpLM  DumiCKHVqoCQKD8roLApzR5Fit8qGV5fVQsJV9sTZk4a    8%   74413454    74413422         86025  1.5.18  13967430.490657268 SOL (4.51%)
592  ChorusmmK7i1AxXeiTtQgQZhQNiXYU84ULeaYF1EH15n  Chorus6Kis8tFHA7AowrPMcRJk3LbApHTYpgSNXzY5KE    8%   74413452    74413420         87344  1.5.18  14992798.131152691 SOL (4.84%)
     Identity                                      Vote Account                            Commission  Last Vote  Root Block Epoch Credits Version  Active Stake

Active Stake: 310011401.560452282 SOL
Current Stake: 309389988.337835133 SOL (99.80%)
Delinquent Stake: 621413.222617161 SOL (0.20%)

Stake By Version:
1.5.14   -   1 current validators ( 0.07%)
1.5.16   -   3 current validators ( 0.07%)
1.5.17   -   1 current validators ( 0.07%)
1.5.18   - 558 current validators (98.46%),   5 delinquent validators ( 0.20%)
1.5.19   -   3 current validators ( 1.14%)
1.5.8    -   0 current validators ( 0.00%),   1 delinquent validators ( 0.00%)
1.6.4    -   1 current validators ( 0.00%)
1.6.7    -   1 current validators ( 0.00%)
unknown  -   1 current validators ( 0.00%),  17 delinquent validators ( 0.00%)
```

Reply:
Perfect!